### PR TITLE
Fix support for window aggregate with non fixed-width accumulator

### PR DESCRIPTION
**Pull request by aditi-pandit**
Required MappedMemory and HashStringAllocator setup for the aggregate objects in the window function.

### CODE REVIEW

File: velox/functions/prestosql/window/AggregateWindow.cpp [hunk 1]
```diff
@@ -54,6 +55,9 @@ class AggregateWindowFunction : public exec::WindowFunction {
     // the function value for each row.
     aggregate_ = exec::Aggregate::create(
         name, core::AggregationNode::Step::kSingle, argTypes_, resultType);
+    // Set HashStringAllocator needed for out of line allocations.
+    stringAllocator_ = std::make_unique<HashStringAllocator>(mappedMemory_);
```
Comment (mbasmanova) [inline velox/functions/prestosql/window/AggregateWindow.cpp:59]: Any particular reason to create new HashStringAllocator for each function? HashAggregation operator creates one HashStringAllocator and passes it to all the aggregate functions.

**Review by mbasmanova — COMMENTED** [review]
@aditi-pandit Looks good. One question.

File: velox/functions/prestosql/window/AggregateWindow.cpp (hunk 1 above)
Comment (aditi-pandit) [reply velox/functions/prestosql/window/AggregateWindow.cpp:59]: I was passing the MappedMemory to all window functions for generality. There are other options we can consider :
i) We need HashStringAllocator only for Aggregate window functions. All other window functions don't need it. Do we use MappedMemory for any other purposes ? If not, then we can change this API to take a HashStringAllocator instead and that could be common for all functions in the Window operator.
ii) We could keep static variables for MappedMemory and HashStringAllocator in the AggregateWindowFunction. So then we could reuse these across aggregates. Of course we will have to change them with each operator. For that we could compare that if the present MappedMemory is different from previous one then to change it. But then these variables would need to be thread-safe also. 
So in general I seem to favor i).
wdyt ?

Comment (mbasmanova) [reply velox/functions/prestosql/window/AggregateWindow.cpp:59]: > we can change this API to take a HashStringAllocator instead

Sounds good.

File: velox/exec/Window.h [hunk 2]
```diff
@@ -145,6 +145,9 @@ class Window : public Operator {
   // The decodedInputVectors_ are reused across addInput() calls to decode
   // the partition and sort keys for the above RowContainer.
   std::vector<DecodedVector> decodedInputVectors_;
+  // HashStringAllocator required by functions that allocate out of line
```
Comment (mbasmanova) [inline velox/exec/Window.h:148]: nit: add empty line between member variables

File: velox/exec/Window.h [hunk 3]
```diff
@@ -145,6 +145,9 @@ class Window : public Operator {
...
+  // string values.
```
Comment (mbasmanova) [inline velox/exec/Window.h:149]: > string values.

Not necessarily "string" values. The name 'HashStringAllocator' is confusing as it reads as it allocates strings. It allocates byte buffers.

**Review by mbasmanova — APPROVED** [review]

**Pull request closed without merging**